docs/ai/how-to/app-service-db-auth.md

```diff
@@ -0,0 +1,310 @@
+```
+
+:::zone-end
+
+### Implement token-based authentication with the vector database
+
+The following code samples require these additional libraries:
+
+* [`Azure.Identity` NuGet package](https://www.nuget.org/packages/Azure.Identity)
+* [`Microsoft.SemanticKernel.Connectors.AzureAISearch` NuGet package](https://www.nuget.org/packages/Microsoft.SemanticKernel.Connectors.AzureAISearch)
+
+1. Initialize a `DefaultAzureCredential` object to pick up your app's managed identity:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="tokenCredential":::
+
+1. Initialize an `IMemoryStore` object for your vector database, then use it to build a `ISemanticTextMemory`:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="aiStore":::
+
+1. Build a `Kernel` object, then import the `ISemanticTextMemory` object using the `TextMemoryPlugin`:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="addMemory":::
+
+1. Use the `Kernel` object to invoke a prompt that includes memory recall:
+
+    :::code language="csharp" source="./snippets/semantic-kernel/IdentityExamples.cs" id="useMemory":::
+
+## Use Key Vault to store connection secrets
+
+If a vector database doesn't support Microsoft Entra authentication, you can use a Key Vault to store your connection secrets and retrieve them with your App Service application. By using a Key Vault to store your connection secrets you can share them with multiple applications, and control access to individual secrets per application.
+
+Before following these steps, retrieve a connection string for your vector database. For example, see [Use Azure Cache for Redis in .NET Framework](/azure/azure-cache-for-redis/cache-dotnet-how-to-use-azure-redis-cache#retrieve-host-name-ports-and-access-keys-from-the-azure-portal).
+
+### Add a connection string to Key Vault
+
+:::zone target="docs" pivot="azure-portal"
+
+Before following these steps, ensure you have [created a Key Vault using the Azure Portal](/azure/key-vault/general/quick-create-portal).
+
+1. Navigate to your key vault in the [Azure Portal](https://aka.ms/azureportal).
+1. In the Key Vault left navigation, select **Objects** then select **Secrets**.
+1. Select **+ Generate/Import**.
+1. On the **Create a secret** screen choose the following values:
+   * **Upload options**: Manual.
+   * **Name**: Type a name for the secret. The secret name must be unique within a Key Vault.
+   * **Value**: The connection string for your vector database
+   * Leave the other values to their defaults. Select **Create**.
+1. Once you receive the message that the secret has been successfully created, it is ready to use in your application.
+
+:::zone-end
+
+:::zone target="docs" pivot="azure-cli"
+
+Before following these steps, ensure you have [created a Key Vault using the Azure CLI](/azure/key-vault/general/quick-create-cli).
```

Comment:
Should this be an alert of some kind?

Reply:
It's mostly just a pre-req for this specific section, same as with the version of these instruction for the Portal

Reply:
I think the 'IMPORTANT' alert style would best represent this info. Since it is necessary for success.